src/backsolve.jl

```diff
@@ -98,8 +152,10 @@ function CuSparseSM(
     A::CUSPARSE.CuSparseMatrixCSR{T}, transa::CUSPARSE.SparseChar, X::CuMatrix{T};
     algo=CUSPARSE.CUSPARSE_SPSM_ALG_DEFAULT,
 ) where T
-    # CUDA 12 is required for inplace computation in cusparseSpSV
-    @assert CUDA.runtime_version() >= v"12.0"
+    chktrans(transa)
+
+    # CUDA 12 is required for inplace computation in cusparseSpSM
+    @assert CUDA.runtime_version() ≥ v"12.0"
```

Comment:
Maybe update to `v"12.3"` as this is required after?

Reply:
:+1: